src/Integration/RuleSetHelper.cs

```diff
@@ -62,6 +63,31 @@ public static IEnumerable<RuleSetInclude> FindAllIncludesUnderRoot(RuleSet ruleS
         }
 
         /// <summary>
+        /// Return a non-nested include from source to target
+        /// </summary>
+        /// <param name="source">Required</param>
+        /// <param name="target">Required</param>
+        /// <returns><see cref="RuleSetInclude"/> or null if not found</returns>
+        public static RuleSetInclude FindInclude(RuleSet source, RuleSet target)
+        {
+            if (source == null)
+            {
+                throw new ArgumentNullException(nameof(source));
+            }
+
+            if (target == null)
+            {
+                throw new ArgumentNullException(nameof(target));
+            }
+
+            string relativeTargetFilePath = PathHelper.CalculateRelativePath(source.FilePath, target.FilePath);
+
+            return source.RuleSetIncludes.SingleOrDefault(i =>
+                StringComparer.OrdinalIgnoreCase.Equals(i.FilePath, relativeTargetFilePath)
```

Comment:
Fair enough. Just something that entered my mind in general when dealing with paths as strings (which IMHO is pretty weak `</RantForTheDay>` :smile: )